fail to make libffi.a on MacOS

when I run 
pyodide % sudo make

It looks fine until it gets to this error
source pyodide/emsdk/emsdk/emsdk_env.sh
make[1]: source: No such file or directory
make[1]: *** [pyodide/cpython/installs/python-3.10.2/lib/libffi.a] Error 1
make: *** [pyodide/cpython/installs/python-3.10.2/lib/python3.10] Error 2

when I check I can see that there is in fact no file called libffi.a in 
pyodide/cpython/installs/python-3.10.2/lib/

Any help would be greatly apprecaited!

I'm also encountering this and I'm running make without sudo. 

The output is below.

```
[2022-04-23 09:34:33] Building cpython...
make -C /Users/michael/Projects/learn-with/pyodide/cpython
rm -rf /Users/michael/Projects/learn-with/pyodide/cpython/build/libffi
git clone https://github.com/hoodmane/libffi-emscripten /Users/michael/Projects/learn-with/pyodide/cpython/build/libffi --shallow-exclude upstream-base
Cloning into '/Users/michael/Projects/learn-with/pyodide/cpython/build/libffi'...
remote: Enumerating objects: 1674, done.
remote: Counting objects: 100% (1674/1674), done.
remote: Compressing objects: 100% (797/797), done.
remote: Total 1674 (delta 1029), reused 1344 (delta 848), pack-reused 0
Receiving objects: 100% (1674/1674), 883.19 KiB | 8.57 MiB/s, done.
Resolving deltas: 100% (1029/1029), done.
source /Users/michael/Projects/learn-with/pyodide/emsdk/emsdk/emsdk_env.sh
make[1]: source: No such file or directory
make[1]: *** [/Users/michael/Projects/learn-with/pyodide/cpython/installs/python-3.10.2/lib/libffi.a] Error 1
make: *** [/Users/michael/Projects/learn-with/pyodide/cpython/installs/python-3.10.2/lib/python3.10] Error 2
```

I'm on an Intel Macbook Pro using zsh. I've checked on sh, bash and zsh and I'm able to manually execute `source` with no problem. 

Switching from zsh to bash or sh, and then running `make` does not change the output.

Attempting to use `./run_docker make` results in a different error:

```
checking whether the C compiler works... no
configure: error: in `/src/cpython/build/libffi':
configure: error: C compiler cannot create executables
See `config.log' for more details
emconfigure: error: './configure --host=wasm32-unknown-linux --prefix=/src/cpython/build/libffi/target --enable-static --disable-shared --disable-dependency-tracking --disable-builddir --disable-multi-os-directory --disable-raw-api --disable-docs' failed (returned 77)
make[1]: *** [Makefile:123: /src/cpython/installs/python-3.10.2/lib/libffi.a] Error 1
make[1]: Leaving directory '/src/cpython'
make: *** [Makefile:229: /src/cpython/installs/python-3.10.2/lib/python3.10] Error 2
```

This also seems related to libffi?

So I hit this and managed to get past it:
```diff
diff --git a/cpython/Makefile b/cpython/Makefile
index 5f069db0..37cc5d37 100644
--- a/cpython/Makefile
+++ b/cpython/Makefile
@@ -119,8 +119,7 @@ $(INSTALL)/lib/libbz2.a: $(BZIP2TARBALL)
 $(INSTALL)/lib/libffi.a :
        rm -rf $(FFIBUILD)
        git clone $(LIBFFIREPO) $(FFIBUILD) --shallow-exclude upstream-base
-   source $(PYODIDE_ROOT)/emsdk/emsdk/emsdk_env.sh
-   cd $(FFIBUILD) && git checkout $(LIBFFI_COMMIT) && ./build.sh && make install
+ . $(PYODIDE_ROOT)/emsdk/emsdk/emsdk_env.sh && cd $(FFIBUILD) && git checkout $(LIBFFI_COMMIT) && pwd && ./build.sh && make install
        cp $(FFIBUILD)/target/include/*.h $(BUILD)/Include/
        mkdir -p $(INSTALL)/lib
        cp $(FFIBUILD)/target/lib/libffi.a $(INSTALL)/lib/
```

It seems [one should not try to `source` in a Makefile](https://stackoverflow.com/a/7508273/119314). Now, after installing `autoconf` from Homebrew, I now get:

```
autoreconf: running: /usr/local/Cellar/autoconf/2.71/bin/autoconf --force
configure.ac:8: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:8: You should run autoupdate.
./lib/autoconf/general.m4:2081: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:8: the top level
configure.ac:74: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:74: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
acinclude.m4:3: AC_FUNC_MMAP_BLACKLIST is expanded from...
configure.ac:74: the top level
configure.ac:91: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:91: You should run autoupdate.
./lib/autoconf/headers.m4:704: AC_HEADER_STDC is expanded from...
configure.ac:91: the top level
configure.ac:118: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:118: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/asmcfi.m4:1: GCC_AS_CFI_PSEUDO_OP is expanded from...
configure.ac:118: the top level
configure.ac:122: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:122: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:122: the top level
configure.ac:138: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:138: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:138: the top level
configure.ac:185: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:185: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:185: the top level
configure.ac:310: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:310: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:310: the top level
configure.ac:418: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:418: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
acinclude.m4:353: LIBFFI_ENABLE_SYMVERS is expanded from...
configure.ac:418: the top level
configure.ac:418: warning: AC_PROG_LD is m4_require'd but not m4_defun'd
acinclude.m4:251: LIBFFI_CHECK_LINKER_FEATURES is expanded from...
acinclude.m4:353: LIBFFI_ENABLE_SYMVERS is expanded from...
configure.ac:418: the top level
configure.ac:418: warning: The macro `AC_TRY_RUN' is obsolete.
configure.ac:418: You should run autoupdate.
./lib/autoconf/general.m4:2997: AC_TRY_RUN is expanded from...
acinclude.m4:251: LIBFFI_CHECK_LINKER_FEATURES is expanded from...
acinclude.m4:353: LIBFFI_ENABLE_SYMVERS is expanded from...
configure.ac:418: the top level
configure.ac:418: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:418: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
acinclude.m4:353: LIBFFI_ENABLE_SYMVERS is expanded from...
configure.ac:418: the top level
configure.ac:41: error: possibly undefined macro: AC_PROG_LIBTOOL
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure:8529: error: possibly undefined macro: AC_PROG_LD
autoreconf: error: /usr/local/Cellar/autoconf/2.71/bin/autoconf failed with exit status: 1
make[1]: *** [/Users/rmay/repos/pyodide/cpython/installs/python-3.10.2/lib/libffi.a] Error 1
make: *** [/Users/rmay/repos/pyodide/cpython/installs/python-3.10.2/lib/python3.10] Error 2
```

Any idea what version of autoconf/automake should work?

Thanks for the heads up. I'm currently relying (I believe) on the native Mac compilers and everything else from Homebrew. I'm stubborn, so I'll keep at it natively. It's now failing with:
```
❯ make
./tools/dependency-check.sh
npx rollup -c src/js/rollup.config.js

./src/js/pyodide.ts → dist/pyodide.mjs...
[!] (plugin Typescript) TS1134: Variable declaration expected.
src/js/pyproxy.gen.ts (136:13)

136 declare var (1 << 8): number;
                ~


make: *** [src/js/_pyodide.out.js] Error 1
```
but I haven't been able to dig it out any further.

--- a/cpython/Makefile
+++ b/cpython/Makefile
@@ -119,8 +119,7 @@ $(INSTALL)/lib/libbz2.a: $(BZIP2TARBALL)
 $(INSTALL)/lib/libffi.a :
 	rm -rf $(FFIBUILD)
 	git clone $(LIBFFIREPO) $(FFIBUILD) --shallow-exclude upstream-base
-	source $(PYODIDE_ROOT)/emsdk/emsdk/emsdk_env.sh
-	cd $(FFIBUILD) && git checkout $(LIBFFI_COMMIT) && ./build.sh && make install
+	. $(PYODIDE_ROOT)/emsdk/emsdk/emsdk_env.sh && cd $(FFIBUILD) && git checkout $(LIBFFI_COMMIT) && pwd && ./build.sh && make install
 	cp $(FFIBUILD)/target/include/*.h $(BUILD)/Include/
 	mkdir -p $(INSTALL)/lib
 	cp $(FFIBUILD)/target/lib/libffi.a $(INSTALL)/lib/

--- a/docs/project/changelog.md
+++ b/docs/project/changelog.md
@@ -14,6 +14,8 @@ substitutions:
 
 ## Unreleased
 
+- {{ Fix }} Fix building on macOS {issue}`2360` {pr}`2554`
+
 - {{ Fix }} Fix a REPL error in printing high-dimensional lists.
   {pr}`2517`
 

